dotnet/src/Connectors/Connectors.Memory.Kusto/KustoExtensions.cs

```diff
@@ -0,0 +1,27 @@
+﻿// Copyright (c) Microsoft. All rights reserved.
+
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Threading;
+using System.Threading.Tasks;
+
+namespace Microsoft.SemanticKernel.Connectors.Memory.Kusto;
+
+internal static class KustoExtensions
+{
+    public static async ValueTask<T?> FirstOrDefaultAsync<T>(this IAsyncEnumerable<T> source, CancellationToken cancellationToken = default)
```

Comment:
Since you're importing the "InternalUtilities" in the csproj:
`  <Import Project="$(RepoRoot)/dotnet/src/InternalUtilities/src/InternalUtilities.props" />`
...you should already have access to this extension from [AsyncEnumerable](https://github.com/microsoft/semantic-kernel/blob/main/dotnet/src/InternalUtilities/src/Linq/AsyncEnumerable.cs) by simply `using System.Linq`. 

Can you try removing this file?

Reply:
@shawncal Good catch! I removed this class and used already existing `AsyncEnumerable` extension method in this commit https://github.com/microsoft/semantic-kernel/pull/2257/commits/3bd036a047feec55bd492ad2b626bf48d1c79551 - it's working as expected.